### PR TITLE
[General Reorganization] Software Stack

### PR DESCRIPTION
**Pull request by yangc95**
### Description

Concept:
Software Stack
Tech Stack* (renamed to software stack) / Full-Stack* (going to request author to make the deletion and addition to software stack post-merge)

Terms:
- Back-End
- Database (not moved in this PR due to existing terms folder)
- Middleware
- Front-End
- Server

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/software-stack/software-stack.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Tags:
+  - 'Development'
+  - 'Frameworks'
+  - 'Rails'
+CatalogContent:
+  - 'learn-rails'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+A **technology (or "tech") stack** refers to a set of software that work together to build digital products or solutions such as websites and mobile applications. Essential for building maintainable and scalable software, tech stacks are also known as technology infrastructures or solution stacks.
```
Comment (caupolicandiaz) [inline content/general/concepts/software-stack/software-stack.md:18]:
```suggestion
A **technology stack** , or "tech" stack, refers to a set of software that work together to build digital products or solutions such as websites and mobile applications. Essential for building maintainable and scalable software, tech stacks are also known as technology infrastructures or solution stacks.
```

File: content/general/concepts/software-stack/software-stack.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+- [Linux](https://www.codecademy.com/resources/docs/open-source/linux): An open-source operating system based on the Linux kernel.
+- Apache: A free and open-source cross-platform web server.
+- MySQL: A relational database management system for storing and manipulating data on the back-end.
+- [PHP](https://www.codecademy.com/resources/docs/php): A server-side scripting programming language used for creating dynamic web applications.
+
+### MEXN Stack
+
+This well-known tech stack comes in a few forms but all of them use:
+
+- [MongoDB](https://www.codecademy.com/learn/learn-mongodb) for the databse. It is a document database used to build highly available and scalable internet applications.
```
Comment (caupolicandiaz) [inline content/general/concepts/software-stack/software-stack.md:49]:
```suggestion
- [MongoDB](https://www.codecademy.com/learn/learn-mongodb) for the database. It is a document database used to build highly available and scalable internet applications.
```

File: content/general/concepts/software-stack/software-stack.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+### Front-End Tech Stack
+
+The [front-end](https://www.codecademy.com/resources/docs/general/software-stack/front-end) is the part of an application the user interacts with. The [UI/UX design](https://www.codecademy.com/resources/docs/uiux) of an application or website should be seamless and straightforward, which is why it is a primary goal of front-end development.
+
+The three major elements that make up the front-end stack are [HTML](https://www.codecademy.com/resources/docs/html), [CSS](https://www.codecademy.com/resources/docs/css), and [JavaScript](https://www.codecademy.com/resources/docs/javascript).
+
+### Back-End Tech Stack
+
+The [back-end](https://www.codecademy.com/resources/docs/general/software-stack/back-end), or server-side, refers to parts of an application's code that allow it to operate and handle new and existing data.It is inaccessible to the user. The back-end of an application is where the data and operating syntax are stored and accessed.
```
Comment (caupolicandiaz) [inline content/general/concepts/software-stack/software-stack.md:28]:
```suggestion
The [back-end](https://www.codecademy.com/resources/docs/general/software-stack/back-end), or server-side, refers to parts of an application's code that allow it to operate and handle new and existing data. It is inaccessible to the user. The back-end of an application is where the data and operating syntax are stored and accessed.
```

**Review by caupolicandiaz — APPROVED** [review]
@yangc95 Is server-side rendering included here as well? Our board currently has it under software development and it wasn't included in the PR description, but the links for server-side rendering have been updated here.

**Review by SSwiniarski — APPROVED** [review]
@yangc95 Looks good once @caupolicandiaz's comments are addressed.

**Comment by yangc95** [comment]
@SSwiniarski @caupolicandiaz 

Thanks for your eyes on this PR

After further consideration, I'm going to close this PR out due to the possibility of other terms that may be beneficial to add under these current concept entries. I'll more than likely create/request a follow-up PR(s) to combine the full/tech-stack concept entries to software stack.

**Pull request closed without merging**